### PR TITLE
Feat/wiki

### PR DESCRIPTION
**Pull request by olastor**
Backend for video wiki

---
This change is [Reviewable](https://reviewable.io/reviews/sirimangalo/meditation-plus-server/122)

### CODE REVIEW

**Comment by olastor** [comment]
```
/home/travis/build/Sirimangalo/meditation-plus-server/src/config/gulpfile.conf.js
  29:23  error  Expected indentation of 2 spaces but found 22  indent
  30:23  error  Expected indentation of 2 spaces but found 22  indent
```
@binarious Do you know how to fix this?

**Comment by binarious** [comment]
Should be fixed in https://github.com/Sirimangalo/meditation-plus-server/pull/122/commits/ce5d6278a94349299732bf02007111253b4ee902.

File: src/app/routes/_wiki.router.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+  /**
+   * Retrieve the Youtube-Video-ID from a URL
+   * @param  {string} url Link to a Youtube-Video
+   * @return {string}     Video-ID of the video (if url type supported by pattern)
+   */
+  function parseVideoID(url) {
+    // source:
+    // stackoverflow.com/questions/3452546/javascript-regex-how-to-get-youtube-video-id-from-url
+    const regExp = /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/;
```
Comment (binarious) [inline src/app/routes/_wiki.router.js:15]: This doesn't support `youtube.com` links. Only `youtu.be`. Is this intentioally?

File: src/app/routes/_wiki.router.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      res.status(500).send(err);
+    }
+  });
+
+  /**
+   * @api {post} /api/wiki Submit a new video or modify an existing one
+   * @apiName SubmitVideo
+   * @apiGroup Wiki
+   */
+  router.post('/api/wiki', async (req, res, next) => {
```
Comment (binarious) [inline src/app/routes/_wiki.router.js:127]: Huge function. Maybe split into smaller ones?

**Review by binarious — CHANGES_REQUESTED** [review]

File: src/app/routes/_wiki.router.js (hunk 1 above)
Comment (olastor) [reply src/app/routes/_wiki.router.js:15]: It should support all sorts of link types. The regexp `youtube\.be` is optional there.
See: http://stackoverflow.com/a/8260383

**Pull request closed without merging**